tests/e2e/test_e2e.py

```diff
@@ -69,13 +71,10 @@ def test_e2e(caplog):
 
     doc_preprocessor = HTMLDocPreprocessor(docs_path, max_docs=max_docs)
 
-    num_docs = session.query(Document).count()
-    if num_docs != max_docs:
-        logger.info("Parsing...")
-        corpus_parser = Parser(
-            structural=True, lingual=True, visual=True, pdf_path=pdf_path
-        )
-        corpus_parser.apply(doc_preprocessor, parallelism=PARALLEL)
+    corpus_parser = Parser(
+        session, structural=True, lingual=True, visual=True, pdf_path=pdf_path
+    )
+    corpus_parser.apply(doc_preprocessor, parallelism=PARALLEL)
```

Comment:
Move parallelism to the constructor?

Reply:
Hmmm, this would restructure all of the UDF classes, but doesn't seem like a bad idea. It would not allow the user to change the parallelism between runs. For example, in our tests, we would need to create new objects to test single vs multithreaded, rather than just being able to change a parameter like we do now.

Reply:
Parallelism isn't required (in that it's a named argument with a default value of 1). I'll take a look at adding it to the constructors.